sympy/polys/matrices/domainmatrix.py

```diff
@@ -1692,6 +1750,582 @@ def det(self):
             raise DMNonSquareMatrixError
         return self.rep.det()
 
+    def adj_det(self):
+        """
+        Adjugate and determinant of a square :class:`DomainMatrix`.
+
+        Returns
+        =======
+
+        (adjugate, determinant) : (DomainMatrix, DomainScalar)
+            The adjugate matrix and determinant of this matrix.
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ
+        >>> from sympy.polys.matrices import DM
+        >>> A = DM([
+        ...     [ZZ(1), ZZ(2)],
+        ...     [ZZ(3), ZZ(4)]], ZZ)
+        >>> adjA, detA = A.adj_det()
+        >>> adjA
+        DomainMatrix([[4, -2], [-3, 1]], (2, 2), ZZ)
+        >>> detA
+        -2
+
+        See Also
+        ========
+
+        adjugate
+        det
+        """
+        m, n = self.shape
+        I_m = self.eye((m, m), self.domain)
+        return self.solve_den_charpoly(I_m, check=False)
+
+    def adjugate(self):
+        """
+        Adjugate of a square :class:`DomainMatrix`.
+
+        The adjugate matrix is the transpose of the cofactor matrix and is
+        related to the inverse by::
+
+            adj(A) = det(A) * A.inv()
+
+        Unlike the inverse matrix the adjugate matrix can be computed and
+        expressed without division or fractions in the ground domain.
+
+        Examples
+        ========
+
+        >>> from sympy import ZZ
+        >>> from sympy.polys.matrices import DM
+        >>> A = DM([[ZZ(1), ZZ(2)], [ZZ(3), ZZ(4)]], ZZ)
+        >>> A.adjugate()
+        DomainMatrix([[4, -2], [-3, 1]], (2, 2), ZZ)
+
+        Returns
+        =======
+
+        DomainMatrix
+            The adjugate matrix of this matrix with the same domain.
+
+        See Also
+        ========
+
+        adj_det
+        """
+        adjA, detA = self.adj_det()
+        return adjA
+
+    def inv_den(self, method=None):
+        """
+        Return the inverse as a :class:`DomainMatrix` with denominator.
+
+        Returns
+        =======
+
+        (inv, den) : (:class:`DomainMatrix`, :class:`~.DomainElement`)
+            The inverse matrix and its denominator.
+
+        Currently this is equivalent to :meth:`adj_det` but it is expected that
+        it might be changed in future. The ratio ``inv/den`` is equivalent to
+        ``adj/det`` except that some factors might be cancelled leading to
+        smaller expressions.
+
+        If the actual adjugate and determinant are needed, use :meth:`adj_det`
+        instead. If the intention is to compute the inverse matrix or solve a
+        system of equations then in future it might be more efficient to use
+        :meth:`inv_den`.
```

Comment:
```suggestion
        system of equations then using :meth:`inv_den` is more efficient.
```

Reply:
I think we can just state this as a fact and leave the definition of future up to an issue that identifies that this method should be used asap instead of `adj_det`.

Reply:
Yes, I agree.

Reply:
I've mentioned the denominator and added see also for other methods.

Reply:
```python
        """Reduced-row echelon form of a with denominator and list of pivots
        See Also
        ========
        sympy.polys.matrices.domainmatrix.DomainMatrix.rref_den
            Higher level interface to this function.
        sympy.polys.matrices.dense.ddm_irref_den
            The underlying algorithm.
```
I don't think I want to add much to docstrings in that module right now because this PR is already big and that module is missing a lot of docstrings.